internal/storage/bucket_handle_test.go

```diff
@@ -405,3 +405,46 @@ func (t *BucketHandleTest) TestListObjectMethodForMaxResult() {
 	AssertEq(TestObjectSubRootFolderName, twoObj.Objects[1].Name)
 	AssertEq(nil, twoObj.CollapsedRuns)
 }
+
+func (t *BucketHandleTest) TestUpdateObjectMethodWithValidObject() {
+	var ContentType string = "ContentType"
+	var ContentEncoding string = "ContentEncoding"
+	var ContentLanguage string = "ContentLanguage"
+	var CacheControl string = "CacheControl"
+
+	_, err := t.bucketHandle.UpdateObject(context.Background(),
+		&gcs.UpdateObjectRequest{
+			Name:                       TestObjectName,
+			Generation:                 TestObjectGeneration,
+			MetaGenerationPrecondition: nil,
+			ContentType:                &ContentType,
+			ContentEncoding:            &ContentEncoding,
+			ContentLanguage:            &ContentLanguage,
+			CacheControl:               &CacheControl,
+			Metadata:                   nil,
+		})
+
+	AssertEq(nil, err)
```

Comment:
you need to compare the object contents as well.

Reply:
commented